Homework 17

testprojp opened the pull request.

testprojp committed:
Create Homework17.java

testprojp committed:
Updated HomeWork17

Updated/added new code for Homework17

Berichicko commented:
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/a605c207-0f3d-4a0b-a8ab-98b9dcb2d2de)

HINT

- It creates a new instance of ChromeOptions, which allows you to configure specific options for the Chrome browser.
- It adds the --remote-allow-origins=* argument to the ChromeOptions, which allows remote origins for websockets. This is done to fix a specific error related to websockets.
- It creates a new instance of ChromeDriver using the ChromeOptions that were set.
- It sets the implicit wait time to 10 seconds using the implicitlyWait method of the driver object. This means that the driver will wait for a certain amount of time when trying to find elements before throwing an exception.
- It maximizes the window using the maximize method of the driver object.
- It calls the urlPage method, which opens the specified URL in the browser.

![image](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/2be83cfc-116a-4e22-9684-4b21f836b307)

testprojp committed:
Updated code

Updated/changed some code in homework17

Berichicko commented:
Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/b1fa731e-89df-4eb8-9ece-4d738c99e4b4)

HINT

1) 

```
public class Homework17 extends BaseTest {
    @Test
    public void addSongToPlaylist() throws InterruptedException {
        String newSongAddedToPlaylistNotification = "Added 1 song into";
        provideEmailCredentials("james.patterson@testpro.io");
        providePasswordCredentials("te$t$tudent");

        loginButton();

        songSearch("Take my Hand");

        viewAllSongs();

        firstSongSelection();

        addToButton();

        selectedSongAddedToPlaylist();
        Assert.assertTrue(addedSongMessage().contains(newSongAddedToPlaylistNotification));
    }
}
```

2) 

![image](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/951ff0c5-9594-4944-b934-112ecf094092)

![image](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/f47b538c-c1a2-463a-bedf-6379fca13762)

For example

![Test status_en](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/f835e3fd-e013-4e24-9f46-fa06607fdc50)

![Test error_en](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/0b9e1c40-239b-4326-bd30-1db5a28aeb7b)

testprojp committed:
Updated homework17

Updated/changed homework17 code "added complete xml tag for playlist selection"

Berichicko commented:
Hello,

I have checked your homework 17, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)

![image](https://github.com/testprojp/A47-QA-Automation-Framework/assets/104761482/3b48721b-14c3-4752-8df0-f90643708828)
